Branch test

kuczynskimaciej opened the pull request.

Commit:
Branch test

Commit:
Próba z ograniczeniem odpowiedzi

sbartnik commented on TestownikConsoleApp/WriteLinesClass.cs:
```diff
@@ -45,20 +47,30 @@ public void WriteLines(string path)
             var selectedAnswers = printAnswers.Where(x => split.Any(y => string.Equals(y, x.Type, StringComparison.InvariantCultureIgnoreCase)));
 
-            if (selectedAnswers.All(x => x.IsCorrect))
+            if (typeOfAnswers.Contains(answer))         //Problem przy porownaniu chara ze stringiem
```
spróbuj z typeOfAnswers.Any(x=> x == answer)

sbartnik commented on TestownikConsoleApp/WriteLinesClass.cs:
```diff
@@ -28,7 +28,9 @@ public void WriteLines(string path)
-            char[] typeOfAnswers = Enumerable.Range('A', 26).Select(x => (char)x).ToArray();
+            int countOfAnswers = file.Skip(2).Count();
+
+            char[] typeOfAnswers = Enumerable.Range('A', countOfAnswers).Select(x => (char)x).ToArray();
```
czemu nie Enumerable.Range('A', listOfAnswers.Length).Select(x => (char)x).ToArray(); ?

Commit:
Próba z ograniczeniem wpisywanych znaków.

Commit:
Komentarze

sbartnik commented on TestownikConsoleApp/WriteLinesClass.cs:
```diff
@@ -56,7 +56,7 @@ public void WriteLines(string path)
             foreach (var y in selectedAnswers)
             {
-                if (result.Contains(y.Type))
+                if (result.Contains(y.Type)) //Robi tyle razy ile jest odpowiedzi przez foreacha, jak to ograniczyc do jednego razu
```
Ogólnie słabe to sprawdzenie. Masz tam obiekt 'split'. Spróbuj poiterować po nim wcześniej i sprawdzić czy w 'typeOfAnswers' jest taka litera, jeżeli nie to wyrzucasz błąd 'Zły znak, wprowadź odpowiedź!'. A jeżeli wszystkie są to pozwalasz mu dalej wykonać kod i później już tego nie sprawdzasz

sbartnik commented on TestownikConsoleApp/WriteLinesClass.cs:
```diff
@@ -47,34 +46,35 @@ public void WriteLines(string path)
 
             Console.Write("Wprowadź odpowiedź: ");
-            string answer = Console.ReadLine();
+            string answer = (string)Console.ReadLine();
```
po co to? zobacz co zwraca ReadLine()

sbartnik commented on TestownikConsoleApp/WriteLinesClass.cs:
```diff
@@ -47,34 +46,35 @@ public void WriteLines(string path)
 
-            if (typeOfAnswers.Contains(answer))         //Problem przy porownaniu chara ze stringiem
-            {
+            string result = new string(typeOfAnswers);
```
wystarczy typeOfAnswers.ToString()

Commit:
Wprowadzono ograniczenia dla wpisanych znaków przy wyborze odpowiedzi.

Wprowadzono sprawdzenie czy plik ma odpowiednią ilość znaczników odpowiedzi do ilości odpowiedzi.

Commit:
Wprowadzono sprawdzenie znaków przy wprowadzeniu odpowiedzi.

sbartnik commented on TestownikConsoleApp/Menu.cs:
```diff
@@ -4,53 +4,68 @@
+                            break;
+                    }
+                    if (number == 0 || number > 2)
+                    {
```
po co to?

sbartnik replied:
użyj 'case default'  czyli wszystko poza 1,2,3 wpadnie do default.

sbartnik commented on TestownikConsoleApp/Menu.cs:
```diff
@@ -4,53 +4,68 @@
+            var caseSwitch = Console.ReadLine();
+            int number;
+
+            bool success = Int32.TryParse(caseSwitch, out number);
```
jak zrobisz case default jak niżej ci napisałem to wtedy ta zmienna success ci nie będzie potrzebna, tylko samo wywołanie Int32.TryParse. Jak nie uda mi się sparsować tego to dostaniesz 0.

sbartnik commented on TestownikConsoleApp/GetPathClass.cs:
```diff
@@ -20,28 +20,24 @@ private Random Generator
-                        int x = this.Generator.Next(0, fileCount);
-                        file = rgFiles.ElementAt(x).FullName;
-                    }
+                    int x = this.Generator.Next(0, fileCount);
```
czym jest x? nazwij to jakoś porządnie.

sbartnik commented on TestownikConsoleApp/Menu.cs:
```diff
@@ -4,53 +4,68 @@
+            string path = getPath.GetRandomFile("Pytania");
 
-            try
+            var caseSwitch = Console.ReadLine();
```
caseSwitch? a może selectedNumber?

sbartnik commented on TestownikConsoleApp/WriteLinesClass.cs:
```diff
@@ -45,24 +47,31 @@ public void WriteLines(string path)
             printAnswers.ForEach(x => Console.WriteLine($"{x.Type}) {x.Answer}"));
```
printAnswers? To je czasownik, czyli opisuje metode. Nazwij to inaczej

sbartnik commented on TestownikConsoleApp/WriteLinesClass.cs:
```diff
@@ -45,24 +47,31 @@ public void WriteLines(string path)
+            string answer = Console.ReadLine();
 
-            var split = answer.Split(',');
+            var split = answer.Split(',').ToList();
```
co ci dał ten ToList tutaj?

Commit:
Poprawki

sbartnik commented on TestownikConsoleApp/GetPath.cs:
```diff
@@ -0,0 +1,44 @@
+
+namespace TestownikConsoleApp
+{
+    class GetPathClass
```
zmieniłeś nazwe pliku ale klasy już nie, i też nazwa klasy nie wydaje ci się dziwna? Bardziej metode przypomina GetPath. Może nazwe warto zmienić na jakiś FileExtension, FileRandomizer?

sbartnik commented on TestownikConsoleApp/Menu.cs:
```diff
@@ -4,21 +4,25 @@
 namespace TestownikConsoleApp
 {
-    class Menu
+    class MenuClass
```
Tu to samo, nazwa pliku ok a po co to "Class'

sbartnik commented on TestownikConsoleApp/Menu.cs:
```diff
@@ -4,21 +4,25 @@
-            int caseSwitch = Convert.ToInt32(Console.ReadLine());
             GetPathClass getPath = new GetPathClass();
-            string path = getPath.getrandomfile("Pytania");
+            string path = getPath.GetRandomFile("Pytania");
```
masz tu przykład :D obiekt 'getPath' na którym wywołujesz metode GetRandomFile

sbartnik commented on TestownikConsoleApp/QuestionDto.cs:
```diff
@@ -4,10 +4,10 @@
 namespace TestownikConsoleApp
 {
-    public class QuestionDto // masz klase AnswerDto którą używasz, zrób tak aby QuestionDTO miało liste AnswerDTO i twoją główną klasą będzie QuestionDTO, CorrectAnswer wywal stąd bo możesz mieć wiele poprawnych odpowiedzi.
+    class QuestionDto
```
o tej klasie możemy pogadać na skype. Obiekt QuestionDto powinien mieć liste AnswerDto i properkę 'Question' w ktorej bedzie treść pytania

slotakonrad replied:
póki co to nie jest nawet nigdzie używane ta klasa.

sbartnik commented on TestownikConsoleApp/Menu.cs:
```diff
@@ -4,21 +4,25 @@
+            var selectedNumber = Console.ReadLine();
+            int number;
+
+            bool success = Int32.TryParse(selectedNumber, out number);
```
ta zmienna w sumie też niepotrzebna, dostaniesz 0 w 'number' kiedy się nie uda sparsować stringa na int, a potem masz tego switcha, gdzie użyłeś default, więc 0 łapie się na default i tam możesz powiedzieć, że 'zły znak'. Pozbędziesz się tym dodatkowego switcha

slotakonrad replied:
to parsowanie w ogóle można wywalić i zastąpić ConsoleKey.D1,  ConsoleKey.D2 itd... Polecam dokumentację :P https://docs.microsoft.com/pl-pl/dotnet/api/system.consolekey?view=netframework-4.8

slotakonrad commented on TestownikConsoleApp/AddNewQuestion.cs:
```diff
@@ -7,12 +7,12 @@ namespace TestownikConsoleApp
             int fileCount = Directory.GetFiles(path, "*.*", SearchOption.AllDirectories).Length;
 
-            var newFile = $@"D:\C#\Programy\TestownikConsoleApp\TestownikConsoleApp\Pytania\{fileCount+1}.txt"; //jakieś gówno
+            
```
enter usuń!

slotakonrad commented on TestownikConsoleApp/AddNewQuestion.cs:
```diff
@@ -7,12 +7,12 @@ namespace TestownikConsoleApp
 {
     class AddNewQuestionClass
```
na co to Class na końcu? czy ta klasa czasami nic nie robi? jak nic to wywalić

slotakonrad commented on TestownikConsoleApp/GetPath.cs:
```diff
@@ -0,0 +1,44 @@
+
+namespace TestownikConsoleApp
+{
+    class GetPathClass
```
Ja bym to raczej nazwał FileService

slotakonrad replied:
i do folderu Services lub Implementations, żeby jakiś porządek był mniej więcej

slotakonrad commented on TestownikConsoleApp/WriteFile.cs:
```diff
@@ -2,21 +2,25 @@
 
 namespace TestownikConsoleApp
 {
     class WriteLinesClass
```
tutaj wiadomo nazwa do zmiany, nazwij to jakoś sensowniej. a co do implementacji. to najpierw zaimplementuj a potem będziemy robić w tej klasie poprawki :P

Commit:
Ograniczenie do wprowadzanych odpowiedzi.

Poprawki.

Commit:
Metoda z tworzeniem nowego pliku.
Poprawki.

sbartnik commented on TestownikConsoleApp/FileReader.cs:
```diff
@@ -2,36 +2,33 @@
 namespace TestownikConsoleApp
 {
-    class WriteLinesClass
+    class ReadFile
```
znowu klasa nazywa się jak jakaś metoda :D w klasie ReadLine masz metode ReadLines :D

sbartnik commented on TestownikConsoleApp/FileReader.cs:
```diff
@@ -42,27 +39,33 @@ public void WriteLines(string path)
 
+            var question = file[1];
+            Console.WriteLine(question);
+            int countOfCorrectAnswers = file[0].Count(x => x == '1');
```
.Count(x => x == '1'); czemu tak? i czemu porównujesz z charem? skoro parametrem(x) w lambdzie Count(x =>) jest int?

Commit:
dupa

Commit:
dupa1

Commit:
Próba rozkminienia - nieudana

Commit:
Próba wypisania pytania

Commit:
Wypisano wszystkie pytania i odpowiedzi

Commit:
Wprowadzono wypisywanie losowego pytania z wymieszanymi odpowiedziami.

slotakonrad commented on TestownikConsoleApp/QuestionWriter.cs:
```diff
@@ -1,35 +1,25 @@
-                {
-                    Console.WriteLine($"{answer.Answer}");
-                });
+                Console.WriteLine($"{answer.Answer}");
```
Tutaj nie jest Ci potrzebny $"" w tym momencie

kuczynskimaciej replied:
Wiem wiem, pozostałości po wczesniejszych próbach.

slotakonrad commented:
można mergować, jest dobrze :D

sbartnik commented on TestownikConsoleApp/QuestionWriter.cs:
```diff
@@ -1,35 +1,25 @@
-
-            questions.ForEach(question =>
+            var countOfQuestions = questions.Count();
+            var indexOfRndQuestion = rnd.Next(countOfQuestions);
```
co się stanie jak będziesz miał np 20 elementów w liście, count of question bedzie wtedy 20 i wylosuje ci 20? A następnie bedziesz wyciagać element o indeksie 20?

kuczynskimaciej replied:
Będzie nadal działać ;)

Commit:
Wypisywanie odpowiedzi 1) randOdp itd.
